src/Widgets/Banner.vala

```diff
@@ -57,11 +57,14 @@ public class AppCenter.Widgets.Banner : Gtk.Button {
         };
         summary_label.get_style_context ().add_class (Granite.STYLE_CLASS_H3_LABEL);
 
-        string[] lines = package.get_description ().split ("\n");
-        var description = lines[0].strip ();
-
-        for (int i = 1; i < lines.length; i++) {
-            description += " " + lines[i].strip ();
+        var description = "";
+        if (package.get_description () != null) {
+            string[] lines = package.get_description ().split ("\n");
+            description = lines[0].strip ();
+
+            for (int i = 1; i < lines.length; i++) {
+                description += " " + lines[i].strip ();
+            }
         }
 
         int close_paragraph_index = description.index_of ("</p>", 0);
```

Comment:
Not directly related to your PR, which seems logical, but I'm not a fan of this `index_of` a thing that might not be there, and the subsequent hardcoded `slice` either.

I think I'd like to have this PR as an opportunity to at least comment why half of this stuff is happening or fix it up